source/core/slang-platform.cpp

```diff
@@ -197,4 +197,52 @@ SLANG_COMPILE_TIME_ASSERT(E_OUTOFMEMORY == SLANG_E_OUT_OF_MEMORY);
 
 #endif // _WIN32
 
+/* static */PlatformKind PlatformUtil::getPlatformKind()
+{
+#if SLANG_WINRT
+    return PlatformKind::WinRT;
+#elif SLANG_XBOXONE
+    return PlatformKind::XBoxOne;
+#elif SLANG_WIN64
+    return PlatformKind::Win64;
+#elif SLANG_X360
+    return PlatformKind::X360;
+#elif SLANG_WIN32
+    return PlatformKind::Win32;
+#elif SLANG_ANDROID
+    return PlatformKind::Android;
+#elif SLANG_LINUX
+    return PlatformKind::Linux;
+#elif SLANG_IOS
+    return PlatformKind::IOS;
+#elif SLANG_OSX
+    return PlatformKind::OSX;
+#elif SLANG_PS3
+    return PlatformKind::PS3;
+#elif SLANG_SLANG_PS4
+    return PlatformKind::PS4;
+#elif SLANG_PSP2
+    return PlatformKind::PSP2;
+#elif SLANG_WIIU
+    return PlatformKind::WIIU;
```

Comment:
I have argued we should use upper camel for acronymns for a variety of reasons. We now have for example GCCCPPCompiler, whereas GccCppCompiler and CppCompiler are much more readable. I know that's not what you are asking here, but if that was the conventions WIIU would naturally be WiiU and we and our eyes could be winners.

Reply:
Ah, but "Wii" isn't an acronym or initialism, so it should be `WiiU` no matter what, right?

On the actual meat of your point... I'm not sure if making `GccCppCompiler` more readable is enough of a win to justify writing stuff like `Aabb` or `D3d12` and so forth. I've already seen in another codebase that *tried* to do the camel-case thing for acronyms/initialisms that people accidentally checked in a bunch of types that didn't fit the rules because the "natural" names were the ones that stuck with all caps.

Maybe the best compromise would be to say "prefer to keep things capitalized because that's how they look most natural, but if things start to look *unnatural* it is okay to switch it up..."

Reply:
`WiiU no matter what, right?` Well depending if it is an acronym or not. One way I have to go figure if it's an acronym to decide, the other I don't. If I do decided it's an acronym then I have to make it less readible. Seems like a double fail ;)
I appreciate the principal of the 'unnatural' get out of jail card, but in practice it's going to have to crazy bad before I go against the convention and have the codebase have two conventions I think. I'll just suffer.